Changed specification of filter function on the joblistings page

miyabitanimchi opened the pull request:
What I did on this branch is as follows:

- Changed the specification of the filter function to and search. It goes to see words on the title, company name, location, short description and tags.
- Implemented sort functions (sorting by employment type and posting date)
- Created a loading component in `components/ui` and inserted it on `components/JobLIstings/jobListings.js`
- Since we need the loading ui, I changed the default states of `jobListingsReducer` and `userJobListingsReducer` to null. I checked the behavior after changing, and checked everything works fine, but please double check. (sorry if I missed some parts to be checked)
- Styled the joblistings page, and also implemented style for smaller device as well.

Please review my code and check the functions when you can, thank you!

BlackStone8960 reviewed (commented):
It looks amazing, thank you so much for doing such a good job.

Could you just apply below changes?
 - Delete search words after searching
 - Make the breakpoint from smartphone to tablet

Thank you.

miyabitanimchi commented:
I made the search value visible on the input field, and also changed the breakpoint mobile to tablet.
Also I added a close icon on the input field so users don't have to delete all the search value with a delete key. I'll also talk with Yusuke san and Shimotaku san about the specification.

Please review again, thank you so much!

BlackStone8960 reviewed (approved):
Looks good, thank you for your quick response :)
